### DISCUSSION TITLE
ETA on 2.4 merge?

### DISCUSSION THREAD

**Discussion by stewcop23** [supermerill/SuperSlicer#2240] [Q&A]
I am checking every day for new updates, I'm on the edge of my seat as it were, but I have no idea when this will be completed. I'm not looking for a specific date, just like a percentage or an estimated date.

**Comment by supermerill** [comment]
I'm going to put a nightly (or beta) tonight.
I think it will be useable, but with many possible badly merged things.

I post the progress in the 'dev' channel of the superslicer discord, if you want more info.

**Comment by tome9111991** [comment]
i am interested in programming c/c++ but unfortunately no time to learn it. 
how difficult is it to merge PS 2.4?

**Reply by supermerill** [reply]
On some files, it's easy, on some it's very time consuming, as each change has to be understood to be written correctly at the right place. Some files are thousand of lines...

From what I said in the discord:
```
Clipper interface has been overhauled (it's the lib to manipulate geometry) many many changes and so many files to update, and some bugfixes to redesign.
Flow has been modified, many lines to change.
Gcode is always a bit long, as it's a huge file.
On this merge, the new big one was the support, as there were big changes. But it wasn't that difficult to merge my changes... I'll see if it's not too buggy.
There were a overhaul of the config file to make it simpler to modify, but so it has been entirely rewritten.
Print & PrintObject have also seen huge changes. For them, it's not possible to 'merge'. I have to get my file, check each change from prusa and integrate it inside, as more than half of the file is in conflict.
Prusa has added ~5 new settings that I already have, but with a different name (I will use the prusa name). The translation will be automatic, but If you open a SuSi 2.4 project in an older verison, you will miss them (and use the default instead).
```

**Comment by neophyl** [comment]
Yeah I didnt think it was going to be quick and easy.  I'm expecting it to take weeks or months.  Theres so many little things , some of which Susi already had and are still better than the prusa version I think.  Like brims.  

Never used Discord, is this a new fangled millennial thing that us old timers dont get ;)
